cl/search/tests/tests_es_recap.py

```diff
@@ -2853,6 +2807,45 @@ def test_cleanup_control_characters_for_xml_rendering(self) -> None:
         with self.captureOnCommitCallbacks(execute=True):
             de_1.delete()
 
+    def test_catch_es_errors(self) -> None:
+        """Can we catch es errors and just render an empy feed?"""
+
+        # Bad syntax error.
+        params = {
+            "q": "Leave /:",
+            "type": SEARCH_TYPES.RECAP,
+        }
+        response = self.client.get(
+            reverse("search_feed", args=["search"]),
+            params,
+        )
+        self.assertEqual(
+            200, response.status_code, msg="Did not get a 200 OK status code."
+        )
```

Comment:
I don't think it's easy, but if it is, we might want to do a custom status code here. Maybe just a 400?

Reply:
Sure, checking options here, I think it'll be easy to write a decorator that can be applied to all the `Feed` views. We can catch the errors on this wrapper and return an HTTP 400. However, we'll lose the empty base feed response.

For instance, currently for the search feed, the base XML is:

```
<?xml version="1.0" encoding="UTF-8"?>
<feed xmlns="http://www.w3.org/2005/Atom" xml:lang="en-us">
   <title>CourtListener.com Custom Search Feed</title>
   <link href="https://www.courtlistener.com/" rel="alternate" />
   <link href="http://example.com/feed/search/" rel="self" />
   <id>https://www.courtlistener.com/</id>
   <updated>2024-03-09T03:20:47.897335+00:00</updated>
   <author>
      <name>Free Law Project</name>
      <email>feeds@courtlistener.com</email>
   </author>
   <rights>Created for the public domain by Free Law Project</rights>
</feed>
```

For Podcast is:
```
<?xml version="1.0" encoding="UTF-8"?>
<rss xmlns:atom="http://www.w3.org/2005/Atom" xmlns:itunes="http://www.itunes.com/dtds/podcast-1.0.dtd" version="2.0">
   <channel>
      <title>CourtListener.com Custom Oral Argument Podcast</title>
      <link>https://www.courtlistener.com/</link>
      <description>A chronological podcast of oral arguments with improved files and meta data. Hosted by Free Law Project through the CourtListener.com initiative. Not an official podcast.</description>
      <atom:link href="http://example.com/podcast/search/" rel="self" />
      <language>en-us</language>
      <copyright>Created for the public domain by Free Law Project</copyright>
      <lastBuildDate>Sat, 09 Mar 2024 03:34:43 +0000</lastBuildDate>
      <itunes:subtitle>A chronological podcast of oral arguments with improved files and meta data. Hosted by Free Law Project through the CourtListener.com initiative. Not an official podcast.</itunes:subtitle>
      <itunes:author>Free Law Project</itunes:author>
      <itunes:summary>A chronological podcast of oral arguments with improved files and meta data. Hosted by Free Law Project through the CourtListener.com initiative. Not an official podcast.</itunes:summary>
      <itunes:explicit>no</itunes:explicit>
      <itunes:owner>
         <itunes:name>Free Law Project</itunes:name>
         <itunes:email>feeds@courtlistener.com</itunes:email>
      </itunes:owner>
      <itunes:image href="https://storage.courtlistener.com/static/png/producer-2000x2000.png" />
      <itunes:category text="Government &amp; Organizations" />
   </channel>
</rss>
```

The metadata also changes a bit if it's a Jurisdiction Feed.
Instead, it'll return a blank response and the HTTP 400.


We could do better and render a default XML for all our feeds in case of an error.
Or we could even render the custom empty feed response while preserving the right metadata for each type of feed. This would require calling the corresponding feed again, but this time with no items. Thus, it would render the base XML for each type of feed with a 400 HTTP code but no feed entries.

Let me know which option seems better to you.

Reply:
I think keeping it simple is best. No need to even do a default feed. Just return a 400 and maybe a textual error if that's easy, but otherwise, if you're using RSS hopefully your reader will throw an error and you can investigate by hand.

Reply:
Perfect. I have applied the refactor. 
Now, the error is caught at the view level, and it returns a `400` status code in the case of a syntax error with the message:
`Invalid search syntax. Please check your request and try again.`

It returns a `500` status code in the case of a server error or network issue with the message:
`Unable to process your request. Please try again later.`